packages/react-instantsearch-dom/src/widgets/__tests__/Answers.js

```diff
@@ -0,0 +1,111 @@
+      Promise.resolve({
+        hits: [{ title: 'hello' }],
+      })
+    ),
+  }),
+});
+
+describe('Answers', () => {
+  let searchClient;
+  beforeEach(() => {
+    searchClient = createSearchClient();
+  });
+
+  it('renders empty Answers widget', () => {
+    const { container } = render(
+      <InstantSearch indexName="ted" searchClient={searchClient}>
+        <Answers searchClient={searchClient} queryLanguages={['en']} />
+      </InstantSearch>
+    );
+    expect(container).toMatchInlineSnapshot(`
+      <div>
+        <div
+          class="ais-Answers ais-Answers--empty"
+        >
+          <div
+            class="ais-Answers-header"
+          />
+          <ul
+            class="ais-Answers-list"
+          />
+        </div>
+      </div>
+    `);
+  });
+
+  it('renders loader when a query is given', () => {
+    const { getByText, getByPlaceholderText } = render(
+      <InstantSearch indexName="ted" searchClient={searchClient}>
+        <SearchBox />
+        <Answers
+          searchClient={searchClient}
+          queryLanguages={['en']}
+          answersComponent={({ isLoading }) =>
+            isLoading ? <p>is loading</p> : <p>not loading</p>
+          }
+        />
+      </InstantSearch>
+    );
+    getByText('not loading');
+    fireEvent.change(getByPlaceholderText('Search here…'), {
+      target: { value: 's' },
+    });
+    getByText('is loading');
+  });
+
+  it('does not render list when loading', () => {
+    const { container, getByPlaceholderText } = render(
+      <InstantSearch indexName="ted" searchClient={searchClient}>
+        <SearchBox />
+        <Answers searchClient={searchClient} queryLanguages={['en']} />
+      </InstantSearch>
+    );
+    fireEvent.change(getByPlaceholderText('Search here…'), {
+      target: { value: 's' },
+    });
+    expect(container.querySelector('ul.ais-Answers-list')).toBeNull();
+  });
+
+  it('renders an answer', async () => {
+    // It's required for `waitFor` method.
+    // https://github.com/testing-library/dom-testing-library/releases/tag/v7.0.0
+    window.MutationObserver = MutationObserver;
+
+    const { getByText, getByPlaceholderText } = render(
+      <InstantSearch indexName="ted" searchClient={searchClient}>
+        <SearchBox />
+        <Answers
+          searchClient={searchClient}
+          queryLanguages={['en']}
+          answersComponent={({ isLoading, hits }) =>
+            !isLoading && hits && hits.length > 0 ? (
+              <div>
+                <p>hits received</p>
+                <pre>{JSON.stringify(hits)}</pre>
+              </div>
+            ) : (
+              <pre>{JSON.stringify({ isLoading, hits })}</pre>
+            )
+          }
+        />
+      </InstantSearch>
+    );
+    expect(() => {
+      getByText('hits received');
+    }).toThrow();
+
+    fireEvent.change(getByPlaceholderText('Search here…'), {
+      target: { value: 'sarah' },
+    });
+    await waitFor(() => getByText('hits received'));
```

Comment:
what about using `runAllMicrotasks` like here

https://github.com/algolia/react-instantsearch/blob/0a711a7533449330cd27b23a17b85b094b4be645/packages/react-instantsearch-core/src/core/__tests__/createInstantSearchManager.derived.js#L3

https://github.com/algolia/react-instantsearch/blob/0a711a7533449330cd27b23a17b85b094b4be645/packages/react-instantsearch-core/src/core/__tests__/createInstantSearchManager.derived.js#L99

(that way you don't need MutationObserver)

Reply:
https://github.com/algolia/react-instantsearch/pull/2996/commits/5dffc128369a87f42197e69ebbb35c99b8d10c70